microsoft/botframework-solutions
# VA Sample deployment script & test updates for nightly build

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*
- Added/fixed ruleset/build.props for VirtualAssistantSample directory
- Added Azure template json to VASample
- Resolve General zh-cn LU model issues
- Sync VA, EA, HA LU files

### Tests
*Is this covered by existing tests or new ones? If no, why not?*
- Update tests to latest `Microsoft.Test.NET.SDK` package
- Updated existing unit tests to `Bot.Builder.LanguageGeneration` package `4.7.1`, requiring a rename of the TemplateEntity singleton to not overstep on the package
- Add TestCategory "UnitTests" to existing test packages
- Add new FunctionalTest project with Direct Line client test scenarios
    - New user greeting
     - Returning user greeting
*This validates bot state and the greeting logic. Note that this is only for English now, we should add additional tests for multiple languages as I've found in bug #2856*
- Add coverlet package in VA/Skill Sample unit test projects to get test status reports in builds

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*
Nightly validation of VA Sample solution
